kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/CommitterImpl.java

```diff
@@ -0,0 +1,253 @@
+import io.tabular.iceberg.connect.events.TopicPartitionOffset;
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import org.apache.iceberg.relocated.com.google.common.annotations.VisibleForTesting;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.util.Pair;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.ListConsumerGroupOffsetsOptions;
+import org.apache.kafka.clients.admin.ListConsumerGroupOffsetsResult;
+import org.apache.kafka.clients.consumer.ConsumerGroupMetadata;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.sink.SinkTaskContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CommitterImpl implements Committer, Closeable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CommitterImpl.class);
+  private final SinkTaskContext context;
+  private final IcebergSinkConfig config;
+  private final boolean isLeader;
+  private final CoordinatorThread coordinatorThread;
+  private final CommitRequestListener commitRequestListener;
+  private final UUID producerId;
+  private final Producer<String, byte[]> producer;
+
+  public CommitterImpl(SinkTaskContext context, IcebergSinkConfig config) {
+    this(
+        context,
+        config,
+        new ConsumerFactoryImpl(),
+        new AdminFactoryImpl(),
+        new ProducerFactoryImpl());
+  }
+
+  private CommitterImpl(
+      SinkTaskContext context,
+      IcebergSinkConfig config,
+      ConsumerFactory consumerFactory,
+      AdminFactory adminFactory,
+      ProducerFactory producerFactory) {
+    this(
+        context,
+        config,
+        new ControlTopicCommitRequestListener(config, consumerFactory),
+        adminFactory,
+        producerFactory,
+        new CoordinatorThreadFactoryImpl(adminFactory, consumerFactory, producerFactory));
+  }
+
+  @VisibleForTesting
+  CommitterImpl(
+      SinkTaskContext context,
+      IcebergSinkConfig config,
+      CommitRequestListener commitRequestListener,
+      AdminFactory adminFactory,
+      ProducerFactory producerFactory,
+      CoordinatorThreadFactory coordinatorThreadFactory) {
+    this.context = context;
+    this.config = config;
+    this.isLeader = Objects.equals(config.taskId(), 0);
+
+    if (isLeader) {
+      this.coordinatorThread = coordinatorThreadFactory.create(config);
+    } else {
+      this.coordinatorThread = null;
+    }
```

Comment:
Previous logic selected whatever task was assigned partition 0, now we choose whatever task=0. 

This might be fine, I wonder if we are guaranteed to always have a task 0?  E.g. in a rebalance is it possible for Task 0's stuff to be rebalanced into Task 1/2 such that there is no Task 0?  From looking at Kafka connect logs over the years I can't recall a case where I've NEVER seen a task 0, so it seems unlikely, just throwing it out there for discussion.

Reply:
You cannot rely on having a task 0, unfortunately (I tried that early on).

Reply:
Ha.  There it is.  KC is always full of surprises.

Reply:
@bryanck can you share more information about this? 
My understanding is that you are saying you Kafka Connect can start task=1 without starting a task=0. 
Are there particular circumstances in which this happens? 
The only way I know of something like this happening is if task=0 fails.

Reply:
Right, task 0 can die and never come back up, leaving you w/out a task 0.

Reply:
Oh, okay phew, that's sane behaviour. 

Ok, so what we're really saying is we want a Coordinator always running somewhere as long as there is at least one task that is alive. Cool, I can't think of a better way of doing that other than the original logic, I'll have to bring that back 👍

Reply:
Reverted back to the original min-topic-partition leader election logic.